### DISCUSSION TITLE
Immich server unable to connect to the database.

### DISCUSSION THREAD

**Discussion by bbrty** [immich-app/immich#14897] [Q&A]
### The bug

Here are the logs from docker, ill paste the logs from immich server. seems to be trying to connect to the DB but failing, it just restarts to fail again. 

`Initializing Immich v1.123.0
Detected CPU Cores: 4
Starting api worker
Starting microservices worker
(node:7) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:7) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:17) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [NestFactory] Starting Nest application...
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] BullModule dependencies initialized +1ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] OpenTelemetryModule dependencies initialized +1ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] OpenTelemetryCoreModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] ClsModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] ScheduleModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] BullModule dependencies initialized +1ms
[Nest] 17  - 12/24/2024, 2:12:30 AM     LOG [InstanceLoader] BullModule dependencies initialized +12ms
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (1)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (2)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (3)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (4)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (5)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (6)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (7)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (8)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (9)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:12:30 AM   ERROR [ExceptionHandler] getaddrinfo ENOTFOUND database
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
api worker exited with code 1
Initializing Immich v1.123.0
Detected CPU Cores: 4
Starting api worker
Starting microservices worker
(node:7) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:17) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:7) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [NestFactory] Starting Nest application...
[Nest] 7  - 12/24/2024, 2:13:17 AM     LOG [NestFactory] Starting Nest application...
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] BullModule dependencies initialized +4ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] OpenTelemetryModule dependencies initialized +1ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] OpenTelemetryCoreModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +1ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] ClsModule dependencies initialized +0ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] ScheduleModule dependencies initialized +1ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] BullModule dependencies initialized +2ms
[Nest] 17  - 12/24/2024, 2:13:17 AM     LOG [InstanceLoader] BullModule dependencies initialized +1ms
[Nest] 17  - 12/24/2024, 2:13:17 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (1)...
Error: getaddrinfo ENOTFOUND database
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)
[Nest] 17  - 12/24/2024, 2:13:17 `

Then heres the log from postgre

`chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted
chown: changing ownership of '/var/lib/postgresql/data': Operation not permitted`

and also redis

'1:C 24 Dec 2024 02:11:21.666 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 24 Dec 2024 02:11:21.666 # Redis version=6.2.16, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 24 Dec 2024 02:11:21.666 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 24 Dec 2024 02:11:21.667 * monotonic clock: POSIX clock_gettime
1:M 24 Dec 2024 02:11:21.668 * Running mode=standalone, port=6379.
1:M 24 Dec 2024 02:11:21.668 # Server initialized
1:M 24 Dec 2024 02:11:21.668 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1:M 24 Dec 2024 02:11:21.669 * Ready to accept connections
1:M 24 Dec 2024 02:16:22.056 * 100 changes in 300 seconds. Saving...
1:M 24 Dec 2024 02:16:22.056 * Background saving started by pid 84
84:C 24 Dec 2024 02:16:22.147 * DB saved on disk
84:C 24 Dec 2024 02:16:22.148 * RDB: 0 MB of memory used by copy-on-write
1:M 24 Dec 2024 02:16:22.157 * Background saving terminated with success
1:M 24 Dec 2024 02:21:23.019 * 100 changes in 300 seconds. Saving...
1:M 24 Dec 2024 02:21:23.020 * Background saving started by pid 153
153:C 24 Dec 2024 02:21:23.088 * DB saved on disk
153:C 24 Dec 2024 02:21:23.090 * RDB: 0 MB of memory used by copy-on-write
1:M 24 Dec 2024 02:21:23.122 * Background saving terminated with success
1:signal-handler (1735007116) Received SIGTERM scheduling shutdown...
1:M 24 Dec 2024 02:25:16.337 # User requested shutdown...
1:M 24 Dec 2024 02:25:16.337 * Saving the final RDB snapshot before exiting.
1:M 24 Dec 2024 02:25:16.839 * DB saved on disk
1:M 24 Dec 2024 02:25:16.839 # Redis is now ready to exit, bye bye...
1:C 24 Dec 2024 02:26:59.893 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 24 Dec 2024 02:26:59.893 # Redis version=6.2.16, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 24 Dec 2024 02:26:59.893 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 24 Dec 2024 02:26:59.894 * monotonic clock: POSIX clock_gettime
1:M 24 Dec 2024 02:26:59.923 * Running mode=standalone, port=6379.
1:M 24 Dec 2024 02:26:59.923 # Server initialized
1:M 24 Dec 2024 02:26:59.923 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1:M 24 Dec 2024 02:26:59.973 * Loading RDB produced by version 6.2.16
1:M 24 Dec 2024 02:26:59.973 * RDB age 103 seconds
1:M 24 Dec 2024 02:26:59.973 * RDB memory usage when created 1.38 Mb
1:M 24 Dec 2024 02:26:59.973 # Done loading RDB, keys loaded: 15, keys expired: 0.
1:M 24 Dec 2024 02:26:59.973 * DB loaded from disk: 0.049 seconds
1:M 24 Dec 2024 02:26:59.973 * Ready to accept connections
1:M 24 Dec 2024 02:32:00.053 * 100 changes in 300 seconds. Saving...
1:M 24 Dec 2024 02:32:00.054 * Background saving started by pid 91
91:C 24 Dec 2024 02:32:00.110 * DB saved on disk
91:C 24 Dec 2024 02:32:00.112 * RDB: 0 MB of memory used by copy-on-write
1:M 24 Dec 2024 02:32:00.154 * Background saving terminated with success
1:M 24 Dec 2024 02:37:01.040 * 100 changes in 300 seconds. Saving...
1:M 24 Dec 2024 02:37:01.041 * Background saving started by pid 161
161:C 24 Dec 2024 02:37:01.104 * DB saved on disk
161:C 24 Dec 2024 02:37:01.106 * RDB: 0 MB of memory used by copy-on-write
1:M 24 Dec 2024 02:37:01.143 * Background saving terminated with success
1:signal-handler (1735007983) Received SIGTERM scheduling shutdown...
1:M 24 Dec 2024 02:39:43.789 # User requested shutdown...
1:M 24 Dec 2024 02:39:43.789 * Saving the final RDB snapshot before exiting.
1:M 24 Dec 2024 02:39:43.824 * DB saved on disk
1:M 24 Dec 2024 02:39:43.824 # Redis is now ready to exit, bye bye...
1:C 24 Dec 2024 02:39:45.736 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 24 Dec 2024 02:39:45.737 # Redis version=6.2.16, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 24 Dec 2024 02:39:45.737 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 24 Dec 2024 02:39:45.738 * monotonic clock: POSIX clock_gettime
1:M 24 Dec 2024 02:39:45.739 * Running mode=standalone, port=6379.
1:M 24 Dec 2024 02:39:45.739 # Server initialized
1:M 24 Dec 2024 02:39:45.739 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1:M 24 Dec 2024 02:39:45.740 * Loading RDB produced by version 6.2.16
1:M 24 Dec 2024 02:39:45.740 * RDB age 2 seconds
1:M 24 Dec 2024 02:39:45.740 * RDB memory usage when created 0.78 Mb
1:M 24 Dec 2024 02:39:45.740 # Done loading RDB, keys loaded: 15, keys expired: 0.
1:M 24 Dec 2024 02:39:45.740 * DB loaded from disk: 0.000 seconds
1:M 24 Dec 2024 02:39:45.740 * Ready to accept connections
'

### The OS that Immich Server is running on

Ubuntu 22.10

### Version of Immich Server

v1.123.0

### Version of Immich Mobile App

v.1.0.0

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
literally the default unedited
```

### Your .env content

```Shell
My .env wouldnt download so i ran docker without it. Is this the cause of the problem, in the logs it said it was using a default .env instead?
```

### Reproduction steps

1.installed onto a mounted USB as storage. Thats literally the only thing i did with my server

**Comment by mmomjian** [comment]
1. you **absolutely** need an ENV, our guides say that and you even got warnings about it not working so why would you expect this to work?
2. NTFS or FAT drives are not supported for the Postgres database
